runtime/lua/vim/lsp/types/protocol.lua

```diff
@@ -0,0 +1,4401 @@
+--[[
+This file is autogenerated from scripts/lsp_types.lua
+Regenerate:
+nvim -l scripts/lsp_types.lua gen --runtime/lua/vim/lsp/types/protocol.lua
+--]]
+
+---@alias lsp.null nil
+---@alias lsp.string string
+---@alias lsp.boolean boolean
+---@alias lsp.integer integer
+---@alias lsp.uinteger integer
+---@alias lsp.decimal number
```

Comment:
Can we directly inline the built-in `integer` / `string`, etc, without an intermediate `lsp.` variant?
Or what's the advantage of the indirection?

Reply:
the advantage is that I don't have to check if the types are any of those ones
I just added `lsp.` to all the types
but it's easily doable to do that
and performance doesn't really matter here anyway since it's so fast and doesn't need to be run often